Improve Neo4jError representation

robsdedude opened the pull request:
When the error is not received from the DBMS, but instead originates from somewhere in the driver, it might not have a code and a message. In that case, we fall back to the default Exception representation.

Related:
 * https://github.com/neo4j/neo4j-python-driver/issues/796

robsdedude committed:
Improve Noe4jError representation

When the error is not received from the DBMS, but instead originates from
somewhere in the driver, it might not have a code and a message. In that case,
we fall back to the default Exception representation.

Related:
 * https://github.com/neo4j/neo4j-python-driver/issues/796

bigmontz commented on neo4j/exceptions.py:
```diff
@@ -247,15 +247,17 @@ def is_fatal_during_discovery(self) -> bool:
 
     def __str__(self):
-        return "{{code: {code}}} {{message: {message}}}".format(code=self.code, message=self.message)
+        if self.code and self.message:
+            return "{{code: {code}}} {{message: {message}}}".format(
+                code=self.code, message=self.message
+            )
+        return super().__str__()
```
Is it possible Neo4jErro be created only with message or code?

robsdedude replied:
Shouldn't happen if the error comes from the server. In that case it goes through the `Neo4jError.hydrate` method, which makes sure both are set to some string. But this is still a valid concern for future-proofing the code.

robsdedude committed:
Make Neo4jError representation more robust

Even though, all errors that come from the server go through
`Noej4jError.hydrate` which (currently) makes sure that `code` *and* `message`
are set, this might change in the future.

bigmontz reviewed (approved):
3️⃣

robsdedude committed:
Merge branch '5.0' into improve-neo4j-error-message
